test/bufconn/bufconn.go

```diff
@@ -171,7 +199,23 @@ func (p *pipe) Write(b []byte) (n int, err error) {
 			if !p.full() {
 				break
 			}
-			p.wwait.Wait()
+			go func() {
```

Comment:
This pattern is duplicated from above.  You may be able to refactor into a helper function.